packages/expect/src/types.ts

```diff
@@ -19,17 +19,21 @@ export type AsyncExpectationResult = Promise<SyncExpectationResult>;
 
 export type ExpectationResult = SyncExpectationResult | AsyncExpectationResult;
 
-export type MatcherFunctionWithState<
-  State extends MatcherState = MatcherState,
+export type MatcherFunctionWithContext<
+  Context extends MatcherContext = MatcherContext,
   Expected extends Array<any> = [] /** TODO should be: extends Array<unknown> = [] */,
-> = (this: State, actual: unknown, ...expected: Expected) => ExpectationResult;
+> = (
+  this: Context,
+  actual: unknown,
+  ...expected: Expected
+) => ExpectationResult;
 
 export type MatcherFunction<Expected extends Array<unknown> = []> =
-  MatcherFunctionWithState<MatcherState, Expected>;
+  MatcherFunctionWithContext<MatcherContext, Expected>;
 
 // TODO should be replaced with `MatcherFunctionWithContext`
```

Comment:
@mrazauskas what is this TODO for?

Reply:
Is I recall it right, `MatcherFunction ` and `MatcherFunctionWithContext` does not allow `any` type of `actual` and `expected` values. Or so. Currently there is a need to have `any`s internally, but would be nicer to rework these into `unknown`s. After this will be done, `RawMatcherFn` can be replaced with `MatcherFunctionWithContext`.

Reply:
In other words, `RawMatcherFn` is redundant type, but it cannot be removed at this moment.